lib/mayaUsd/render/MaterialXGenOgsXml/GlslFragmentGenerator.cpp

```diff
@@ -385,8 +385,45 @@ ShaderPtr GlslFragmentGenerator::generate(
                 true);
         }
 
-        // Add all function calls
+        // Add all function calls (varies greatly by MaterialX version)
```

Comment:
Nothing complex here. We have a local version of the GlslGenerator that closely follows the original one. There were some missing updates that are visible by comparing the same lines for MaterialX [v1.38.2](https://github.com/AcademySoftwareFoundation/MaterialX/blob/v1.38.2/source/MaterialXGenGlsl/GlslShaderGenerator.cpp#L594), [1.38.3](https://github.com/AcademySoftwareFoundation/MaterialX/blob/v1.38.3/source/MaterialXGenGlsl/GlslShaderGenerator.cpp#L603), and [1.38.5](https://github.com/AcademySoftwareFoundation/MaterialX/blob/v1.38.5/source/MaterialXGenGlsl/GlslShaderGenerator.cpp#L639).

These versions were copied over and bracketed with the right conditional MX version compiler checks.